### ISSUE TITLE
False positive: file_get_contents() on an uploaded file (UploadedFile) reports TaintedFile + TaintedSSRF

### ISSUE THREAD

**Issue by alies-dev** [psalm/laravel-psalm-plugin#1134]
## Summary

Reading an uploaded file with `file_get_contents($request->file('x'))` — an extremely common, safe Laravel idiom — raises two false-positive taint criticals: `TaintedFile` and `TaintedSSRF`.

## Reproduction

Found while auditing [Bottelet/DaybydayCRM](https://github.com/Bottelet/DaybydayCRM).

```php
// controller
$file = $request->file('file');                 // Illuminate\Http\UploadedFile
$fileData = $fileSystem->upload($folder, $filename, $file);

// service
public function upload($folder, $filename, $file): array
{
    $file_content = file_get_contents($file);   // <-- TaintedFile + TaintedSSRF reported here
    ...
}
```

Taint trace (abbreviated):
```
Illuminate\Http\Request::file   (stubs/common/Http/Request.phpstub:54, @psalm-taint-source input)
  -> $file = $request->file('file')
  -> upload($folder, $filename, $file)
  -> file_get_contents($file)     sink
```

## Why it's a false positive

`file()` returns an `UploadedFile`. Passing it to `file_get_contents()` coerces it via
`SplFileInfo::__toString()` to `getPathname()` — the PHP-generated temp upload path in
`upload_tmp_dir`. That path is:

- not attacker-controllable (PHP assigns a random temp name; the request cannot set it),
- never a URL (`file()` only reads the uploaded-files bag; a text field returns `null`, not a string), so SSRF via `file_get_contents` is structurally impossible here,
- not redirectable to arbitrary files (no LFI).

The attacker controls the file **content** and the **client original name**, neither of which is the path argument to `file_get_contents`. So the path-based sinks (`TaintedFile`, `TaintedSSRF`) should not fire.

## Root cause

`stubs/common/Http/Request.phpstub` marks `file()` with a coarse `@psalm-taint-source input`. That taints the whole `UploadedFile`, including its `__toString`/path coercion, so any path sink it reaches lights up. The content is user input; the temp **path** is not.

## Suggested fix

Model `UploadedFile` taint more precisely so the path coercion does not feed file/SSRF sinks while the content stays tainted. Options:

- Add an `UploadedFile` stub whose `__toString()`/`getPathname()`/`getRealPath()` return values are **not** tainted (sandboxed temp path), while `getContent()`/`get()` and `getClientOriginalName()` remain tainted sources.
- Or scope the `file()` source so its taint attaches to content accessors rather than the path coercion.

## Impact

Reading an upload via `file_get_contents()` (or passing the temp path to other path APIs) is a standard Laravel pattern. As-is, every such app gets two false `Tainted*` criticals, which erodes trust in the security output and bloats baselines.
